projects/core/src/lib/layers/legend/legend-raster/raster-legend.component.ts

```diff
@@ -87,4 +87,17 @@ export class RasterLegendComponent implements OnInit, OnChanges {
             this.numberPipeParameters = calculateNumberPipeParameters(symbology.colorizer.getBreakpoints());
         }
     }
+
+    shortenDecimals(toShorten: number): number {
+        const DECIMAL_PLACES = 3; // number of decimal places left after rounding
+        const integerPart: number = Math.floor(toShorten);
+        const decimals: number = toShorten - integerPart;
+        const decimalsString = decimals.toString();
+        let zeroes = 0;
+        while (decimalsString.charAt(zeroes + 2) === '0') {
+            zeroes++;
+        }
+        const roundAt: number = Math.pow(10, zeroes + DECIMAL_PLACES);
+        return integerPart + Math.round(decimals * roundAt) / roundAt;
+    }
```

Comment:
I don't think this is a good heuristic for the use case.
You have to consider all values that you want to display.
Then, when you want to distinguish them, you need to add decimal places to do so.

Also, you want to be kind of consistent with all the numbers to once specify one decimal place and other times six.

I was wondering now if we did not tackle this problem before and did some research.
Please take a look at https://github.com/geo-engine/geoengine-ui/pull/158/files.
I'm not quite sure where this code went.

Then, like Karo did back then, please add a test for different arrays of breakpoints to see that they are all calculated properly.

Reply:
Here is what I came up with:
There might be a case when values differ only in their last few digits, but are otherwise digits other than zero, for example 3.1111112345 and 3.1111119876. This probably happens very rarely, but in theory it might happen. So right now I'm first comparing all the values to see if there is any overlap. If so, rounding must take place 3 decimal places behind the last index where all values overlap. Otherwise I'm checking for the greatest amount of zeroes, which then becomes the minimum amount of places for all numbers, before any rounding takes place.
The larger of these two values takes precedence.

A worst case scenario would be when two values are like the above examples, but the others are all entirely different values (4, 5 etc), so those two would be flattened to 3.111... but I guess that can't happen since breakpoints are evenly distributed?

The code still needs some cleanup and refactoring, and I still need to add the tests (tested my methods in a separate environments for simplicity) but perhaps you could give me some feedback on whether I'm on the right track.

Reply:
In general, you can cut off after the last digit which helps you to distinguish between the values.

If you have significant gaps, for instance, 1,000,000.23345435 to 5,000,000.4365464, you can, in principle, cut all decimal places. On the other hand, if you have values distributed between 3.1111112345 and 3.1111119876, you must not make them indistinguishable by outputting [3, 3].

As stated above, please write some tests for different cases so that you can see the effects.